sound/soc/sof/intel/cnl.c

```diff
@@ -228,6 +229,22 @@ void cnl_ipc_dump(struct snd_sof_dev *sdev)
 		hipcida, hipctdr, hipcctl);
 }
 
+static int cnl_register_clients(struct snd_sof_dev *sdev)
+{
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_PROBES)
+	return sof_client_dev_register(sdev, "hda-probes", 0);
+#else
+	return 0;
+#endif
```

Comment:
Now that I think of it, would there is any harm is creating a client device unconditionally?

If there is no driver registered and the auxiliary bus doesn't find a match, then nothing will happen.
We should really make use of the Linux device/driver model.

@ranj063 any comments?